lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -4,163 +4,507 @@
-    const component = mount(
-      <NxTable>
-        <NxTableHead>
-          <NxTableRow>
-            <NxTableCell>Foo</NxTableCell>
-            <NxTableCell>Bar</NxTableCell>
-          </NxTableRow>
-        </NxTableHead>
-        <NxTableBody error="Error!"/>
-      </NxTable>,
-      { attachTo: mountContainer }
-    );
+          expect(screen.getByRole('button')).toHaveAccessibleName('dolphin');
+        });
+      });
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderColumnHeader({
+            chevron: true,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          })!;
+
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+    });
+
+    describe('when the rowBtnIcon is set', function() {
+      const renderTableCell = (extraProps?: NxTableCellProps, isHeader?: boolean) => {
+        const children = (
+          <NxTable.Row isClickable={true} clickAccessibleLabel="dolphin">
+            <NxTable.Cell>Foo</NxTable.Cell>
+            <NxTable.Cell { ...extraProps } />
+          </NxTable.Row>
+        );
+
+        return render(
+          <NxTable>
+            { isHeader ? <NxTable.Head>{children}</NxTable.Head>
+            : <NxTable.Body>{children}</NxTable.Body>
+            }
+          </NxTable>
+        ).getAllByRole(isHeader ? 'columnheader' : 'cell')[1] as HTMLElement;
+      };
+
+      describe('when isHeader', function() {
+        it('renders a header cell with the text "Select Row"', function() {
+          const component = renderTableCell({
+            rowBtnIcon: faEdit,
+            sortDir: 'asc',
+            children: <span>foo</span>
+          }, true)!;
 
-    await act(async () => {
-      component.setProps({
-        children: (
-          <>
-            <NxTableHead>
-              <NxTableRow>
-                <NxTableCell>Foo</NxTableCell>
-              </NxTableRow>
-            </NxTableHead>
-            <NxTableBody error="Error!"/>
-          </>
-        )
+          expect(component).toHaveTextContent('Select Row');
+        });
+      });
+
+      describe('when not isHeader', function() {
+        it('does not show button when rowBtnIcon is undefined', function() {
+          expect(within(renderTableCell({ rowBtnIcon: undefined })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('does not show button when rowBtnIcon is null', function() {
+          expect(within(renderTableCell({ rowBtnIcon: null })).queryByRole('button')).not.toBeInTheDocument();
+        });
+
+        it('show button when rowBtnIcon faEdit is set', function() {
+          expect(within(renderTableCell({ rowBtnIcon: faEdit })).getByRole('button')).toBeInTheDocument();
+        });
+
+        it('ignores the children and sort settings and adds the icon child wrapped in a nx-cell__row-btn',
```

Comment:
`.nx-cell__row-btn` is no longer involved in this test

Reply:
> `.nx-cell__row-btn` is no longer involved in this test

Cleaned up here: cec50dbad93958d240cdfabd25323834588f180f